src/accounts/biconomy/signerToBiconomySmartAccount.ts

```diff
@@ -0,0 +1,389 @@
+                type: "address"
+            }
+        ],
+        stateMutability: "nonpayable",
+        type: "function"
+    }
+] as const
+
+/**
+ * Default addresses for Biconomy Smart Account
+ */
+const BICONOMY_ADDRESSES: {
+    ECDSA_OWNERSHIP_REGISTRY_MODULE: Address
+    ACCOUNT_V2_0_LOGIC: Address
+    FACTORY_ADDRESS: Address
+    DEFAULT_FALLBACK_HANDLER_ADDRESS: Address
+} = {
+    ECDSA_OWNERSHIP_REGISTRY_MODULE:
+        "0x0000001c5b32F37F5beA87BDD5374eB2aC54eA8e",
+    ACCOUNT_V2_0_LOGIC: "0x0000002512019Dafb59528B82CB92D3c5D2423aC",
+    FACTORY_ADDRESS: "0x000000a56Aaca3e9a4C479ea6b6CD0DbcB6634F5",
+    DEFAULT_FALLBACK_HANDLER_ADDRESS:
+        "0x0bBa6d96BD616BedC6BFaa341742FD43c60b83C1"
+}
+
+const BICONOMY_PROXY_CREATION_CODE =
+    "0x6080346100aa57601f61012038819003918201601f19168301916001600160401b038311848410176100af578084926020946040528339810103126100aa57516001600160a01b0381168082036100aa5715610065573055604051605a90816100c68239f35b60405162461bcd60e51b815260206004820152601e60248201527f496e76616c696420696d706c656d656e746174696f6e206164647265737300006044820152606490fd5b600080fd5b634e487b7160e01b600052604160045260246000fdfe608060405230546000808092368280378136915af43d82803e156020573d90f35b3d90fdfea2646970667358221220a03b18dce0be0b4c9afe58a9eb85c35205e2cf087da098bbf1d23945bf89496064736f6c63430008110033"
+
+/**
+ * Get the account initialization code for Biconomy smart account with ECDSA as default authorization module
+ * @param owner
+ * @param index
+ * @param factoryAddress
+ * @param ecdsaValidatorAddress
+ */
+const getAccountInitCode = async ({
+    owner,
+    index,
+    factoryAddress,
+    ecdsaModuleAddress
+}: {
+    owner: Address
+    index: bigint
+    factoryAddress: Address
+    ecdsaModuleAddress: Address
+}): Promise<Hex> => {
+    if (!owner) throw new Error("Owner account not found")
+
+    // Build the module setup data
+    const ecdsaOwnershipInitData = encodeFunctionData({
+        abi: BiconomyInitAbi,
+        functionName: "initForSmartAccount",
+        args: [owner]
+    })
+
+    // Build the account init code
+    return concatHex([
+        factoryAddress,
+        encodeFunctionData({
+            abi: createAccountAbi,
+            functionName: "deployCounterFactualAccount",
+            args: [ecdsaModuleAddress, ecdsaOwnershipInitData, index]
+        }) as Hex
+    ])
+}
+
+const getAccountAddress = async <
+    TTransport extends Transport = Transport,
+    TChain extends Chain | undefined = Chain | undefined
+>({
+    client,
+    factoryAddress,
+    accountLogicAddress,
+    fallbackHandlerAddress,
+    ecdsaModuleAddress,
+    entryPoint,
+    owner,
+    index = 0n
+}: {
+    client: Client<TTransport, TChain>
+    factoryAddress: Address
+    accountLogicAddress: Address
+    fallbackHandlerAddress: Address
+    ecdsaModuleAddress: Address
+    owner: Address
+    entryPoint: Address
+    index?: bigint
+}): Promise<Address> => {
+    // @ts-ignore
```

Comment:
need to resolve this.

Reply:
currently I don't see use of entryPoint or client but we may require client. hence I have kept it to be consistent with rest of interfaces

Reply:
for now I have removed these and fixed.